coordinate-systems.md

```diff
@@ -0,0 +1,246 @@
+
+So is XRFrameOfReference (FOR) is really an XRCoordinateSystem? It doesn't share the same definition as a standard 3D coordinate system - it doesn't represent a transform in a shared space that can be used to track relationships between different objects in the world. Really, it just represents the current style of determining where the device is in 3D space. For example, objects would have the same transform in _eye-level_ and _head-model_ and _stage_, it is only the **device** transform that changes. If I get up and move around in _head-model_ mode, the position of the device doesn't change and thus my position relative to objects in the world does not change - only the relationship between my orientation and the orientation of objects changes. Similar for the stage FOR - if I am playing a VR game where I teleport around the world using my stage, the translation of my device is being manipulated in order to artificially change its position relative to teleportation behavior. This naturally changes my relationship to objects in the world, as, when I teleport, an object that was previously 10 meters away might now be right in front of me. What matters here is that while the transform of the device can be represented by an XRCoordinateSystem, it is merely the nature of the translation component of that coordinate system and thus the virtual camera that changes based on which FOR mode is being used.
+
+Possible Proposal: Instead of a class that inherits from XRCoordinateSystem, XRFrameOfReference could simply be an enumeration of different styles of representing the device/input-source poses in 3D space:
+
+
+```
+const devicePose = xrFrame.getDevicePose('stage');
+```
+
+
+Possible Proposal: FOR could even be a configuration option on the XRSession or a parameter to the call to get the device transform:
+
+
+```
+device.requestSession({ frameOfReference: "stage", outputContext: ctx });
+or
+xrSession.setDevicePoseMode("stage");
+and then
+const devicePose = xrFrame.getDevicePose();
+```
+
+
+We wouldn't need methods that convert XRCoordinateSystems to different FORs, as the positions of all devices and input-sources are always expressed implicity in terms of the current FOR.
+
+Note: Another important thing of note is that XRFrameOfReference currently contains the stage bounds and estimated height. These will need to be moved to another API for access, probably directly on the XRSession since they are not dynamic within a session. This change should probably happen regardless as it is weird having these properties tacked on to a class inheriting from XRCoordinateSystem anyway.
+
```

Comment:
Another alternative here would be an `XRSession.setDefaultCoordinateSystem` method.  This would allow apps to omit passing the base coordinate system to an API if your app wishes to use its normal default coordinate system.

We could go further then and actually omit the coordinate system parameters from those APIs, but this would limit the expressiveness of the app.  For example, an app could not query for the relative transform between a pair of meshes or planes that were detected previously if the device just lost tracking, since those planes can not be related to the app's chosen frame of reference for rendering.  The API should be able to answer honestly an app's requests about world geometry, without assuming the only purpose of the request is to directly facilitate rendering this frame.